communication/src/main/java/org/jboss/da/communication/pom/PomAnalyzerImpl.java

```diff
@@ -80,54 +80,50 @@ public GAVDependencyTree readRelationships(File pomRepoDir, File pomPath,
                 Map<File, ProjectVersionRef> projectVersionRefs = getProjectVersionRefs(tempDir,
                         findAllPomFiles(pomRepoDir));
 
-                ProjectVersionRef rootProject = null;
-
                 // for each pom.xml
                 for (Map.Entry<File, ProjectVersionRef> entry : projectVersionRefs.entrySet()) {
-                    // if pomPath is the same as what we found analysis the root repository
-                    if (pomPath.getCanonicalPath().equals(entry.getKey().getCanonicalPath())) {
-                        rootProject = entry.getValue();
-                        break;
+                    Set<ProjectRelationship<?, ?>> relationships;
+                    try {
+                        // find all the dependencies / plugins / parents / plugins etc for each pom.xml
+                        relationships = getDiscoveryResult(tempDir, pomRepoDir, entry.getValue(),
+                                repositories).getAcceptedRelationships();
```

Comment:
@jbartece Initially analyzing only the requested module/Pom seemed to make sense. Which is what the previous logic does. But analyzing the `application` folder in dependency-analysis only shows the modules that `application` pom.xml uses, not the modules dependencies... since we never analyzed the modules dependencies.